src/app/beer_garden/db/mongo/models.py

```diff
@@ -358,8 +364,21 @@ class Request(MongoModel, Document):
 
     logger = logging.getLogger(__name__)
 
+    def pre_serialize(self):
+        """If string output was over 16MB it was spilled over to the GridFS storage solution"""
+        if self.output_gridfs:
+            self.output = self.output_gridfs.read().decode("utf-8")
+            self.output_gridfs = None
+
     def save(self, *args, **kwargs):
         self.updated_at = datetime.datetime.utcnow()
+
+        # If the output size is too large, we switch it over
+        # Max size for Mongo is 16MB, switching over at 15MB to be safe
+        if self.output and sys.getsizeof(self.output) < (1000000 * 15):
```

Comment:
How was this demonstrated to work? This needs to be `>`.

Reply:
hmm.... You are correct..... I don't know why I have it backwards....